lua/packer/plugin_utils.lua

```diff
@@ -100,26 +100,43 @@ plugin_utils.ensure_dirs = function()
 end
 
 plugin_utils.find_missing_plugins = function(plugins, opt_plugins, start_plugins)
-  if opt_plugins == nil or start_plugins == nil then
-    opt_plugins, start_plugins = plugin_utils.list_installed_plugins()
-  end
+  return a.sync(function ()
+    if opt_plugins == nil or start_plugins == nil then
+      opt_plugins, start_plugins = plugin_utils.list_installed_plugins()
+    end
 
-  -- NOTE/TODO: In the case of a plugin gaining/losing an alias, this will force a clean and
-  -- reinstall
-  local missing_plugins = {}
-  for _, plugin_name in ipairs(vim.tbl_keys(plugins)) do
-    local plugin = plugins[plugin_name]
+    -- NOTE/TODO: In the case of a plugin gaining/losing an alias, this will force a clean and
+    -- reinstall
+    local missing_plugins = {}
+    for _, plugin_name in ipairs(vim.tbl_keys(plugins)) do
+      local plugin = plugins[plugin_name]
 
-    local plugin_path = util.join_paths(config[plugin.opt and 'opt_dir' or 'start_dir'],
-                                        plugin.short_name)
-    local plugin_installed = (plugin.opt and opt_plugins or start_plugins)[plugin_path]
+      local plugin_path = util.join_paths(config[plugin.opt and 'opt_dir' or 'start_dir'],
+      plugin.short_name)
+      local plugin_installed = (plugin.opt and opt_plugins or start_plugins)[plugin_path]
 
-    if not plugin_installed or plugin.type ~= plugin_utils.guess_dir_type(plugin_path) then
-      table.insert(missing_plugins, plugin_name)
+      await(vim.schedule)
```

Comment:
Nitpick: Let's use `await(a.main)` for consistency. It's the same thing, but in case we have to change the async implementation in the future, I'd prefer to use `a.main`.